src/rpc/rpc_client.cpp

```diff
@@ -3,248 +3,278 @@
-        std::vector<rpc::Block> blocks;
-        blocks.resize(n);
-        for (size_t i = 0; i < n; ++i) {
-            blocks[i] = response.blocks(i);
-        }
-        return {blocks};
-    }
-    return {};
+op_string RPCClient::GetVertex(std::string block_hash) {
+    GetVertexRequest request;
+    request.set_hash(block_hash);
+
+    GetVertexResponse response;
+    return ProcessResponse(
+        [&](auto* context, auto request, auto* response) -> grpc::Status {
+            return be_stub_->GetVertex(context, request, response);
+        },
+        request, &response);
+}
+
+op_string RPCClient::GetForks() {
+    EmptyRequest request;
+    GetForksResponse response;
+    return ProcessResponse(
+        [&](auto* context, auto request, auto* response) -> grpc::Status {
+            return be_stub_->GetForks(context, request, response);
+        },
+        request, &response);
+}
+
+op_string RPCClient::GetPeerChains() {
+    EmptyRequest request;
+    GetPeerChainsResponse response;
+    return ProcessResponse(
+        [&](auto* context, auto request, auto* response) -> grpc::Status {
+            return be_stub_->GetPeerChains(context, request, response);
+        },
+        request, &response);
+}
+
+op_string RPCClient::GetRecentStat() {
+    EmptyRequest request;
+    GetRecentStatResponse response;
+    return ProcessResponse(
+        [&](auto* context, auto request, auto* response) -> grpc::Status {
+            return be_stub_->GetRecentStat(context, request, response);
+        },
+        request, &response);
+}
+
+op_string RPCClient::Statistic() {
+    EmptyRequest request;
+    StatisticResponse response;
+    return ProcessResponse(
+        [&](auto* context, auto request, auto* response) -> grpc::Status {
+            return be_stub_->Statistic(context, request, response);
+        },
+        request, &response);
 }
 
-std::optional<StatusResponse> RPCClient::Status() {
-    StatusRequest request;
+op_string RPCClient::Status() {
+    EmptyRequest request;
     StatusResponse response;
-    grpc::ClientContext context;
-    grpc::Status status = commander_stub_->Status(&context, request, &response);
-    if (!status.ok()) {
-        spdlog::error("No response from RPC server: {}", status.error_message());
-        return {};
-    }
-    return {response};
+    return ProcessResponse(
+        [&](auto* context, auto request, auto* response) -> grpc::Status {
+            return commander_stub_->Status(context, request, response);
+        },
+        request, &response);
 }
 
 bool RPCClient::Stop() {
-    StopRequest request;
+    EmptyRequest request;
     StopResponse response;
     grpc::ClientContext context;
     grpc::Status status = commander_stub_->Stop(&context, request, &response);
     if (!status.ok()) {
-        spdlog::error("No response from RPC server: {}", status.error_message());
+        std::cout << "No response from RPC server: " << status.error_message() << std::endl;
```

Comment:
Why don't we use spdlog and why don't we use ProcessResponse? The return type can also be templated.

Reply:
First, we have decided not to include spdlog on rpc client side. Then this method only returns a bool but not optional string. There is no need to use the template. For the rest of methods returning optional string but not using `ProcessResponse`, changes will be make in the PR about return code.

Reply:
Fine